pkg/azure/api/api.go

```diff
@@ -351,6 +351,8 @@ func parseSubnet(subnet *network.Subnet) (s *ipamTypes.Subnet) {
 		s.CIDR = c
 		if subnet.IPConfigurations != nil {
 			s.AvailableAddresses = c.AvailableIPs() - len(*subnet.IPConfigurations)
+		} else {
+			s.AvailableAddresses = c.AvailableIPs()
```

Comment:
The change makes sense, maybe add a comment with a not explaining the motivation for future reference?